scripts/tasks/check-and-rewrite-package-json.js

```diff
@@ -139,6 +129,6 @@ if (differingPackageJsonFiles.length > 0) {
     // Only in "test" mode do we actually emit a non-zero exit code. This is designed for CI tests.
     // In other cases (e.g. as a git precommit hook), we can just exit with a normal 0 exit code.
     if (process.argv.includes('--test')) {
-        process.exit(1);
+        process.exitCode = 1;
```

Comment:
Is there a big difference here?

Reply:
Not really, just best practice:
> In most situations, it is not actually necessary to call `process.exit()` explicitly. The Node.js process will exit on its own _if there is no additional work pending_ in the event loop. The `process.exitCode` property can be set to tell the process which exit code to use when the process exits gracefully.
>
> For instance, the following example illustrates a _misuse_ of the `process.exit()` method that could lead to data printed to stdout being truncated and lost:
>
> ```js
>import { exit } from 'node:process';
>
>// This is an example of what *not* to do:
>if (someConditionNotMet()) {
>  printUsageToStdout();
>  exit(1);
>}
>```
> The reason this is problematic is because writes to `process.stdout` in Node.js are sometimes asynchronous and may occur over multiple ticks of the Node.js event loop. Calling `process.exit()`, however, forces the process to exit before those additional writes to `stdout` can be performed.
>
> Rather than calling `process.exit()` directly, the code _should_ set the `process.exitCode` and allow the process to exit naturally by avoiding scheduling any additional work for the event loop:
>
>```js
>import process from 'node:process';
>
>// How to properly set the exit code while letting
>// the process exit gracefully.
>if (someConditionNotMet()) {
>  printUsageToStdout();
>  process.exitCode = 1;
>}
>```
> If it is necessary to terminate the Node.js process due to an error condition, throwing an uncaught error and allowing the process to terminate accordingly is safer than calling _process.exit()_.

https://nodejs.org/docs/latest-v22.x/api/process.html#processexitcode

Reply:
TIL! Thanks.